api/message.go

```diff
@@ -329,16 +331,58 @@ func (a *MessageAPI) DeleteMessage(ctx *gin.Context) {
 //     schema:
 //         $ref: "#/definitions/Error"
 func (a *MessageAPI) CreateMessage(ctx *gin.Context) {
-	message := model.Message{}
+	message := model.MessageExternal{}
 	if err := ctx.Bind(&message); err == nil {
 		application := a.DB.GetApplicationByToken(auth.GetTokenID(ctx))
 		message.ApplicationID = application.ID
 		if strings.TrimSpace(message.Title) == "" {
 			message.Title = application.Name
 		}
 		message.Date = timeNow()
-		a.DB.CreateMessage(&message)
-		a.Notifier.Notify(auth.GetUserID(ctx), &message)
+		msgInternal := toInternalMessage(&message)
+		a.DB.CreateMessage(msgInternal)
+		a.Notifier.Notify(auth.GetUserID(ctx), toExternalMessage(msgInternal))
 		ctx.JSON(200, message)
```

Comment:
If we do not use the internal message, then the ID will always be 0 on the returned message.
```suggestion
		ctx.JSON(200, toExternalMessage(msgInternal))
```